fuel/src/test/kotlin/com/github/kittinunf/fuel/ReadmeIntegrityTest.kt

```diff
@@ -0,0 +1,244 @@
+import com.github.kittinunf.fuel.core.Method
+import com.github.kittinunf.fuel.core.extensions.jsonBody
+import com.github.kittinunf.fuel.test.MockHttpTestCase
+import com.github.kittinunf.fuel.test.MockReflected
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.notNullValue
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+class ReadmeIntegrityTest : MockHttpTestCase() {
+
+    // This silences the printing so it doesn't pollute the log
+    private val outContent = ByteArrayOutputStream()
+    private val originalOut = System.out
+
+    @Before
+    fun prepareStream() {
+        System.setOut(PrintStream(outContent))
+        Fuel.reset()
+    }
+
+    @After
+    fun teardownStream() {
+        System.setOut(originalOut)
+    }
+
+    @Test
+    fun makingRequestsExample() {
+        reflectedRequest(Method.GET, "get")
+            .response { request, response, result ->
+                println("[request] $request")
+                println("[response] $response")
+                val (bytes, error) = result
+                if (bytes != null) {
+                    println("[response bytes] ${String(bytes)}")
+                }
+
+                assertThat("Expected bytes, actual error $error", bytes, notNullValue())
+            }
+            .join()
+    }
+
+    @Test
+    fun makingRequestsAboutPatchRequests() {
+        mock.chain(
+            request = mock.request().withMethod(Method.POST.value),
+            response = mock.reflect()
+        )
+
+        Fuel.patch(mock.path("/post"))
+            .also { println("[request] $it") }
+            .response()
+    }
+
+    @Test
+    fun makingRequestsAddingParametersGet() {
+        val parameters = listOf("foo" to "foo", "bar" to "bar")
+        val endpoint = "https://httpbin.org/get"
+
+        val request = Fuel.get(endpoint, parameters = parameters)
+        assertThat(request.url.toString(), equalTo("$endpoint?foo=foo&bar=bar"))
+        val request2 = endpoint.httpGet(parameters = parameters)
+        assertThat(request2.url.toString(), equalTo("$endpoint?foo=foo&bar=bar"))
+    }
+
+    @Test
+    fun makingRequestsAddingParametersDelete() {
+        val parameters = listOf("foo" to "foo", "bar" to "bar")
+        val endpoint = "https://httpbin.org/delete"
+
+        val request = Fuel.delete(endpoint, parameters = parameters)
+        assertThat(request.url.toString(), equalTo("$endpoint?foo=foo&bar=bar"))
+        val request2 = endpoint.httpDelete(parameters = parameters)
+        assertThat(request2.url.toString(), equalTo("$endpoint?foo=foo&bar=bar"))
+    }
+
+    @Test
+    fun makingRequestsAddingParametersGetArray() {
+        val parameters = listOf("foo" to "foo", "dwarf" to arrayOf("grumpy", "happy", "sleepy", "dopey"))
```

Comment:
@SleeplessByte  who swaps `Lionel Richie` for the seven dwarfs?

Reply:
:')